src/components/projectconfig/configmgr.cpp

```diff
@@ -0,0 +1,399 @@
+#include "configmgr.h"
+#include <QtCore/QDir>
+#include <QtCore/QFile>
+#include <QtCore/QJsonArray>
+#include <QtCore/QJsonDocument>
+#include <QtCore/QJsonObject>
+
+#define MODULID_MASK 0xFF
+
+ConfigMgr::ConfigMgr()
+{
+}
+
+bool ConfigMgr::getAndCheckId(const QJsonObject& json, unsigned int& id)
+{
+    if (checkData(json, "Id", QJsonValue::Double) == false) {
+        return false;
+    }
+
+    id = json["Id"].toInt();
+    // Check range
+    if ((id <= Config::ModulType::MT_NULL) || (id >= Config::ModulType::MT_MAX)) {
+        cds_error("ConfigMgr::getAndCheckId - Id is out of range");
+        return false;
+    }
+    return true;
+}
+
+bool ConfigMgr::getAndCheckProperties(const QJsonObject& json, QJsonArray& arr)
+{
+    if (json.contains("properties") == false) {
```

Comment:
Why not use checkData() also here, to reduce the code duplication? 

Longer term: It looks like there is even more to do to reduce duplication in all this checking code.  If the project wants to check JSON input against a schema, maybe an even more generic approach is better.  I.e. write a data structure (for example in JSON, or otherwise in C++) that defines all the required fields and their types, etc.  Then loop through it and validate the whole input.  Just a suggestion.

Reply:
@gunnarx Validation with JSON schema was added recently in #93 We still need to create better schema file with "required" fields though...

Reply:
Great to hear it @rkollataj.  Then my comment can be ignored for now :-)